driver-core/src/test/unit/com/mongodb/internal/async/function/RetryStateTest.java

```diff
@@ -260,77 +177,14 @@ void breakAndCompleteIfRetryAndPredicateThrows(final TimeoutContext timeoutConte
-                        (e1, e2) -> expectedTimeoutException,
-                        (rs, e) -> false));
-
-        Assertions.assertEquals(actualTimeoutException, expectedTimeoutException);
-    }
-
-    @Test
-    @DisplayName("should throw timeout exception from retry, when transformer swallows original timeout exception")
-    void advanceThrowTimeoutExceptionWhenTransformerSwallowOriginalTimeoutException() {
-        RetryState retryState = new RetryState(TIMEOUT_CONTEXT_INFINITE_GLOBAL_TIMEOUT);
-        RuntimeException previousAttemptException = new RuntimeException() {
-        };
-        MongoOperationTimeoutException expectedTimeoutException = TimeoutContext.createMongoTimeoutException("Server selection failed");
-
-        retryState.advanceOrThrow(previousAttemptException,
-                (e1, e2) -> previousAttemptException,
-                (rs, e) -> true);
-
-        MongoOperationTimeoutException actualTimeoutException =
-                assertThrows(expectedTimeoutException.getClass(), () -> retryState.advanceOrThrow(expectedTimeoutException,
-                        (e1, e2) -> previousAttemptException,
-                        (rs, e) -> false));
-
-        Assertions.assertNotEquals(actualTimeoutException, expectedTimeoutException);
-        Assertions.assertEquals("Retry attempt timed out.", actualTimeoutException.getMessage());
-        Assertions.assertEquals(previousAttemptException, actualTimeoutException.getCause(),
-                "Retry timeout exception should have a cause if transformer returned non-timeout exception.");
-    }
-
-
-    @Test
-    @DisplayName("should throw original timeout exception from retry, when transformer returns original timeout exception")
-    void advanceThrowOriginalTimeoutExceptionWhenTransformerReturnsOriginalTimeoutException() {
-        RetryState retryState = new RetryState(TIMEOUT_CONTEXT_INFINITE_GLOBAL_TIMEOUT);
-        RuntimeException previousAttemptException = new RuntimeException() {
-        };
-        MongoOperationTimeoutException expectedTimeoutException = TimeoutContext
-                .createMongoTimeoutException("Server selection failed");
-
-        retryState.advanceOrThrow(previousAttemptException,
-                (e1, e2) -> previousAttemptException,
-                (rs, e) -> true);
-
-        MongoOperationTimeoutException actualTimeoutException =
-                assertThrows(expectedTimeoutException.getClass(), () -> retryState.advanceOrThrow(expectedTimeoutException,
-                        (e1, e2) -> expectedTimeoutException,
-                        (rs, e) -> false));
-
-        Assertions.assertEquals(actualTimeoutException, expectedTimeoutException);
-        Assertions.assertNull(actualTimeoutException.getCause(),
-                "Original timeout exception should not have a cause if transformer already returned timeout exception.");
-    }
-
```

Comment:
Note the removal of the above 3 tests. I have removed all references to timeouts from RetryState and these tests: RetryState should now not be aware of timeouts, and should have no special behaviours associated with timeouts. Please confirm that this is expected.

Reply:
Reverted